src/Pages/SinglePage/SinglePage.js

```diff
@@ -0,0 +1,238 @@
+import "./SinglePage.css";
+import { BatNetworkContainer } from "../../Components/Wrapper/BatNetworkContainer";
+import { Sidebar } from "../../Components/Sidebar/Sidebar";
+import { RightSidebar } from "../../Components/RightSidebar/RightSidebar";
+import { useParams, useNavigate } from "react-router-dom";
+import { Avatar } from "../../Components/Avatar/Avatar";
+import { useSelector, useDispatch } from "react-redux";
+import { getAllUsers } from "../../features/user";
+
+import {
+  getSinglePost,
+  CommentCard,
+  likePost,
+  dislikePost,
+} from "../../features/post";
+import { useEffect, useState } from "react";
+import { ButtonPrimary } from "../../Components/Buttons";
+import {
+  UserListModal,
+  addBookMark,
+  removeBookMark,
+} from "../../features/user";
+
+const SinglePage = () => {
+  const dispatch = useDispatch();
+  const navigate = useNavigate();
+  const { postId } = useParams();
+  const { singlePost, posts, isLoading } = useSelector((state) => state.post);
+  const { users } = useSelector((state) => state.user);
+  const token = useSelector((state) => state.auth.token);
+  const bookmarks = useSelector((state) => state.user.bookmarks);
+  const user = useSelector((state) => state.auth.user);
+  const currentPost = posts?.find((post) => post?.id === postId);
+
+  const isPostInBookMarks = bookmarks?.find(
+    (bookmark) => bookmark?._id === currentPost?._id
+  );
+
+  const currentUser = users?.find(
+    (dbUser) => dbUser.username === singlePost?.username
+  );
+
+  const loggedInUser = users?.find(
+    (dbUser) => dbUser.username === user.username
+  );
+
+  const [comment, setComment] = useState({ reply: "" });
+
+  const commentChangeHandler = (e) => {
+    const { name, value } = e.target;
+    setComment((prevState) => ({ ...prevState, [name]: value }));
+  };
+
+  useEffect(() => {
+    dispatch(getSinglePost(currentPost._id));
+    dispatch(getAllUsers());
+  }, [dispatch, posts, currentPost._id]);
+
+  const [userModal, setUserModal] = useState({
+    show: false,
+    title: "",
+    list: [],
+  });
+
+  const isLikedByLoggedInUser = currentPost?.likes.likedBy?.find(
+    (likeUser) => likeUser.username === user.username
+  );
```

Comment:
Check using a unique ID instead of the username. There are chances of the username being the same for different users and might yield incorrect results.